### PR TITLE
[backport] Fix SI-7862: MANIFEST.MF file for Scala sources

### PR DESCRIPTION
**Pull request by dragos**
This is a backport of PR #2970. In order to move the build to Scala OSGi bundles, we need to be able to build both on 2.10 and 2.11.

In order to be able to use published Scala jars as OSGi bundles in the
Eclipse build, Eclipse needs to match sources and binaries. That is
done by making source jars _source bundles_. This PR adds the required
manifest entries. Nothing else should be affected (file names remain
the same).

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -1886,13 +1886,52 @@ TODO:
     <jar whenmanifestonly="fail" destfile="${dist.dir}/src/scala-library-src.jar">
       <fileset dir="${src.dir}/library"/>
       <fileset dir="${src.dir}/continuations/library"/>
+      <manifest>
+        <attribute name="Manifest-Version" value="1.0"/>
+        <attribute name="Bundle-Name" value="Scala Standard Library Sources"/>
+        <attribute name="Bundle-SymbolicName" value="org.scala-lang.scala-library.source"/>
+        <attribute name="Bundle-Version" value="${osgi.version.number}"/>
+        <attribute name="Eclipse-SourceBundle" value="org.scala-lang.scala-library;version=&quot;${osgi.version.number}&quot;;roots:=&quot;.&quot;" />
+      </manifest>
```
Comment (retronym) [inline build.xml:1895]: I'd like to see this collapsed down to:

```
<macrodef name="osgi.source.manifest">...</>

<osgi.source.manifest name="scala-library" desc="Scala Standard Library"/>
```

Comment (retronym) [inline build.xml:1895]: And at the place you define that macro, add a short comment echoing your commit comment.

Comment (dragos) [inline build.xml:1895]: I always wanted to learn about ant macros!

**Comment by jsuereth** [comment]
LGTM modulo @retronym's statements.

**Comment by dragos** [comment]
See PR #2970. I'll re-open the backport once that one is merged.

**Pull request closed without merging**